evgeniyamelone/QA_Automation_Course
# My store page object

Implement an automated test using TestNG + Selenium for new account registration on http://automationpractice.com
Submit a link to the pull request (dev branch to master) with the completed test to this assignment.

Test scenario:
 
1.  Navigate to the target automationpractice.com; 
2. Open registration form; 
3. Fill all fields in the form with the valid data; 
4. Click 'Register'; 
5. Verify that new account is successfully created. 

Additional requirements:
 
* Target browser: Chrome 
* The test should be stable and may be executed multiple times without any false failures.


- page classes containing functionality for interacting with the page and verify data on it (use PageFactory pattern)
- class for storing account data 
- logging (slf4j)
- configuration property file to store the parameters required for the test 

Implement following improvements for your test framework:

- install Lombok (plugin and dependency) and apply Lombok annotations to your model class; 
- store test data in JSON data files; 
- simple data pool class to handle data files; 
- implement login\logout parameters for the test scenario 

Tests Implementation 
Implement tests for:
 - verification of account data for created account ( personal info and addresses ); 
- edit personal info and addresses.
